rmc/src/lib.rs

```diff
@@ -105,30 +107,43 @@ impl<T> Debug for DoublingDelayScheduler<T> {
 }
 
 impl<T> DoublingDelayScheduler<T> {
-    pub fn new(initial_delay: time::Duration) -> Self {
-        let (on_new_task_tx, on_new_task_rx) = unbounded();
+    pub fn new(initial_delay: Duration) -> Self {
         DoublingDelayScheduler {
             initial_delay,
             scheduled_instants: BinaryHeap::new(),
             scheduled_tasks: Vec::new(),
-            on_new_task_tx,
-            on_new_task_rx,
         }
     }
+
+    /// Choose `delta >> 0` to prevent task bunching.
+    pub fn new_with_tasks(initial_tasks: Vec<T>, initial_delay: Duration, delta: Duration) -> Self {
```

Comment:
Also, constructors usually call each other in a different manner – in this case it would be more common to have this constructor do the whole construction, and `new` call this one with an empty `Vec` of tasks. It could also just be called `with_tasks`.

Reply:
Not sure if we want a split which depends on `initial_delay`. I'd say we expect the newly added tasks to be triggered kind of immediately, while `initial_delay` is potentially large.

Reply:
aight, done